static/app/views/dashboardsV2/widgetBuilder/buildStepYAxisOrColumns.tsx

```diff
@@ -0,0 +1,135 @@
+import styled from '@emotion/styled';
+import cloneDeep from 'lodash/cloneDeep';
+
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization, TagCollection} from 'sentry/types';
+import {
+  explodeField,
+  generateFieldAsString,
+  getAggregateAlias,
+  QueryFieldValue,
+} from 'sentry/utils/discover/fields';
+import Measurements from 'sentry/utils/measurements/measurements';
+import {SPAN_OP_BREAKDOWN_FIELDS} from 'sentry/utils/performance/spanOperationBreakdowns/constants';
+import ColumnEditCollection from 'sentry/views/eventsV2/table/columnEditCollection';
+import {generateFieldOptions} from 'sentry/views/eventsV2/utils';
+import Field from 'sentry/views/settings/components/forms/field';
+
+import {WidgetQuery, WidgetType} from '../types';
+
+import BuildStep from './buildStep';
+import {DataSet, DisplayType} from './utils';
+
+interface Props {
+  dataSet: DataSet;
+  displayType: DisplayType;
+  onChange: (queryIndex: number, newQuery: WidgetQuery) => void;
+  organization: Organization;
+  queries: WidgetQuery[];
+  tags: TagCollection;
+  widgetType: WidgetType;
+  errors?: Record<string, string>[];
+}
+
+export function BuildStepYAxisOrColumns({
```

Comment:
Should we split this out into separate components for Columns and YAxis?
We can then let the WidgetBuilder component handle the logic of hiding and showing the right steps so that it's not obscured by this component?